backend/corpora/lambdas/api/v1/common.py

```diff
@@ -35,3 +39,13 @@ def delete_dataset_common(db_session: Session, dataset: Dataset, token_info: dic
                 original.create_revision(dataset.collection.id)  # Restore the original dataset and S3 assets
             dataset.asset_deletion()  # Delete the S3 assets and database rows.
             dataset.delete()  # Delete the dataset row.
+
+
+def validate_dataset_identifier(db_session, collection_uuid, dataset_uuid, curator_tag) -> Dataset:
```

Comment:
this doesn't really describe (to me) what this function does.

How about `get_dataset_by_identifier`?

Reply:
Good refactor though to consolidate the shared logic 👍

Reply:
+1 on both of dan's comments

Reply:
change name to get_dataset_else_error